pandas/core/arrays/integer.py

```diff
@@ -287,7 +288,8 @@ def __init__(self, values, mask, copy=False):
                 and is_integer_dtype(values.dtype)):
             raise TypeError("values should be integer numpy array. Use "
                             "the 'integer_array' function instead")
-        if not (isinstance(mask, np.ndarray) and is_bool_dtype(mask.dtype)):
```

Comment:
Maybe just remove the isinstance check entirely? Perhaps `is_bool_dtype` is enough.

Reply:
we have a guarantee that an array-like mask is the only acceptable mask (e.g. a list is not acceptable)

Reply:
Lists can't be considered True in `is_bool_dtype`, since they don't have a dtype.

Reply:
right, actually i think will work if I don't do ``mask.dtype``, just ``mask``